Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -0,0 +1,370 @@
+            .arg('-NoLogo')
+            .arg('-NoProfile')
+            .arg('-NonInteractive')
+            .arg('-ExecutionPolicy')
+            .arg('Unrestricted')
+            .arg('-Command')
+            .arg(`. '${this._scriptPath.replace("'", "''")}'`);
+        return tool;
+    }
+
+    public async cleanUp(): Promise<void> {
+        await Util.deleteFile(this._scriptPath);
+    }
+}
+
+class PowerShell6 extends ScriptType {
+
+    public async getTool(): Promise<any> {
+        this._scriptPath = await Util.getPowerShellScriptPath(this._scriptLocation, 'ps1', this._scriptArguments);
+        let tool: any = tl.tool(tl.which('pwsh', true))
+            .arg('-NoLogo')
+            .arg('-NoProfile')
+            .arg('-NonInteractive')
+            .arg('-ExecutionPolicy')
+            .arg('Unrestricted')
+            .arg('-Command')
+            .arg(`. '${this._scriptPath.replace("'", "''")}'`);
+        return tool;
+    }
+
+    public async cleanUp(): Promise<void> {
+        await Util.deleteFile(this._scriptPath);
+    }
+
+}
+
+class Bash extends ScriptType {
+
+    public async getTool(): Promise<any> {
+        this._scriptPath = await Util.getScriptPath(this._scriptLocation, 'sh');
+        let tool: any = tl.tool(tl.which("bash", true));
+        tool.arg(this._scriptPath);
+        tool.line(this._scriptArguments); // additional scriptArguments should always call line. line() parses quoted arg strings
+        return tool;
+    }
+}
+
+class Batch extends ScriptType {
+
+    public async getTool(): Promise<any> {
+        this._scriptPath = await Util.getScriptPath(this._scriptLocation, 'bat');
+        let tool: any = tl.tool(tl.which(this._scriptPath, true));
+        tool.line(this._scriptArguments); // additional scriptArguments should always call line. line() parses quoted arg strings
+        return tool;
+    }
+}
+
+
+class Util {
+
+    public static async getScriptPath(scriptLocation:string, fileExtension:string): Promise<string> {
+        if (scriptLocation === "scriptPath") {
+            let filePath: string = tl.getPathInput("scriptPath", true, true);
+            if(Util.checkIfFileExists(filePath, fileExtension)){
+                return filePath;
+            }
+            throw new Error(tl.loc('JS_InvalidFilePath', filePath));
+        }
+        let tmpDir = tl.getVariable('Agent.TempDirectory') || os.tmpdir();
+        let inlineScript: string = tl.getInput("inlineScript", true);
+        let scriptPath: string = path.join(tmpDir, `azureclitaskscript${new Date().getTime()}.${fileExtension}`);
+        await Util.createFile(scriptPath, inlineScript);
+        return scriptPath;
+    }
+
+    public static async getPowerShellScriptPath(scriptLocation: string, fileExtension: string, scriptArguments: string): Promise<string> {
+        let powerShellErrorActionPreference: string = tl.getInput('powerShellErrorActionPreference', false) || 'Stop';
```

Comment:
if this input is not specified, should we not ignore error action rather than defaulting to 'stop'?